crates/api/src/sitemap.rs

```diff
@@ -0,0 +1,50 @@
+use actix_web::{
+  http::header::{self, CacheDirective},
+  web::Data,
+  HttpResponse,
+};
+use chrono::{DateTime, FixedOffset};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_db_schema::{source::post::Post, utils::DbPool};
+use lemmy_utils::error::LemmyResult;
+use sitemap_rs::{url::Url, url_set::UrlSet};
+use tracing::info;
+
+async fn generate_urlset(pool: &mut DbPool<'_>, limit: i64) -> LemmyResult<UrlSet> {
```

Comment:
The limit for the sitemap shouldn't be in the GET request, is that correct? IE its always a constant?

In that case I would remove this param, and simply hardcode the limit in the `list_for_sitemap` function in the Post impl.

Reply:
I think in general the limit could be configurable via query params, but I have yet to see an example of this actually being implemented by anyone. Let's keep it simple for now and just hard code it to 50000 in the `list_for_sitemap` function.